compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -87,6 +87,12 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             } else if lang_items.sized_trait() == Some(def_id) {
                 // Sized is never implementable by end-users, it is
                 // always automatically computed.
+
+                // Error type cannot possibly implement `Sized` (fixes #123154)
+                if let ty::Error(_) = obligation.predicate.skip_binder().self_ty().kind() {
```

Comment:
Probably better to use `error_reported` instead of `self_ty().kind` to pick up on any errors in the predicate. Not that it matters for Sized, but we'll also find errors in generic args or other aggregates this way

Reply:
Okay

Reply:
Have stopped the ICE. The duplicate diagnostics are also gone. but it has lead to to https://github.com/rust-lang/rust/issues/68830 regressing. I'm investigating.